python/cudf/cudf/_lib/copying.pyx

```diff
@@ -822,12 +823,10 @@ cdef class _CPackedColumns:
 
     def serialize(self):
         header = {}
-        frames = []
+        frames = [Buffer(self.gpu_data_ptr, self.gpu_data_size, self)]
```

Comment:
Would it be better to directly call `Buffer.serialize()` (and subsequently `Buffer.deserialize()` below) ?

Reply:
I think we need to leave this as a `Buffer` so we are able to inherit the additional methods of `Serializable`; see this assertation in `Serializable.device_serialize()`:

https://github.com/rapidsai/cudf/blob/5b8895dc50a5daf99bafede67617e8ebd39d3af6/python/cudf/cudf/core/abc.py#L98-L100

Reply:
Sorry if this is a dumb question, could you elaborate: why `Buffer.deserialize()` fails to inherit the additional methods?

Reply:
Ah my mistake I was thinking of just passing the output of `Buffer.deserialize()` into `frames`, which wouldn't work because of the assertion above - you are correct, we should be serializing the `Buffer` here, and add the resulting header to the `PackedColumns` header.